stdlib/typing_extensions.pyi

```diff
@@ -569,8 +562,70 @@ else:
     ReadOnly: _SpecialForm
     TypeIs: _SpecialForm
 
+# TypeAliasType was added in Python 3.12, but had significant changes in 3.14.
+if sys.version_info >= (3, 14):
+    from typing import TypeAliasType as TypeAliasType
+else:
+    @final
+    class TypeAliasType:
+        def __init__(
+            self, name: str, value: Any, *, type_params: tuple[TypeVar | ParamSpec | TypeVarTuple, ...] = ()
+        ) -> None: ...  # value is a type expression
+        @property
+        def __value__(self) -> Any: ...  # a type expression
+        @property
+        def __type_params__(self) -> tuple[TypeVar | ParamSpec | TypeVarTuple, ...]: ...
+        @property
+        def __parameters__(self) -> tuple[TypeVar | ParamSpec, ...]: ...
+        @property
+        def __name__(self) -> str: ...
+        # It's writable on types, but not on instances of TypeAliasType.
+        @property
+        def __module__(self) -> str | None: ...  # type: ignore[override]
+        # Returns typing._GenericAlias, which isn't stubbed.
+        def __getitem__(self, parameters: Incomplete | tuple[Incomplete, ...]) -> Any: ...
+        def __init_subclass__(cls, *args: Unused, **kwargs: Unused) -> NoReturn: ...
+        def __call__(self) -> NoReturn: ...
```

Comment:
I think including this is likely to mean that type checkers start inferring `TypeAliasType` instances as being callable, which is technically true but not very useful since `__call__` always raises. It's true that they'll (hopefully) issue dead-code warnings for the lines of code after calls to instances of `TypeAliasType`, but dead-code warnings from type checkers are often opt-in rather than opt-out. Overall I think it'll be a much better user experience for users of type checkers if we omit this method from the stub.